platform/osgi/platform-osgi-condpermadmin/src/main/java/org/codice/ddf/condpermadmin/PermissionActivator.java

```diff
@@ -0,0 +1,205 @@
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details. A copy of the GNU Lesser General Public
+ * License is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.ddf.condpermadmin;
+
+import java.io.File;
+import java.io.FileReader;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+import net.sourceforge.prograde.policy.SecurityActions;
+import net.sourceforge.prograde.policyparser.ParsedPermission;
+import net.sourceforge.prograde.policyparser.ParsedPolicy;
+import net.sourceforge.prograde.policyparser.ParsedPolicyEntry;
+import net.sourceforge.prograde.policyparser.ParsedPrincipal;
+import net.sourceforge.prograde.policyparser.Parser;
+import net.sourceforge.prograde.type.Priority;
+import org.apache.commons.lang3.text.StrSubstitutor;
+import org.osgi.framework.BundleActivator;
+import org.osgi.framework.BundleContext;
+import org.osgi.service.condpermadmin.ConditionInfo;
+import org.osgi.service.condpermadmin.ConditionalPermissionAdmin;
+import org.osgi.service.condpermadmin.ConditionalPermissionInfo;
+import org.osgi.service.condpermadmin.ConditionalPermissionUpdate;
+import org.osgi.service.permissionadmin.PermissionInfo;
+import org.osgi.util.tracker.ServiceTracker;
+
+/** Initializes the CondPermAdmin permission table from all policy files in ddf_home/security */
+public class PermissionActivator implements BundleActivator {
+
+  private static final String BUNDLE_NAME_CONDITION =
+      "org.codice.ddf.condition.BundleNameCondition";
+
+  private static final String PRINCIPAL_CONDITION = "org.codice.ddf.condition.PrincipalCondition";
+
+  private static final String SIGNER_CONDITION = "org.codice.ddf.condition.SignerCondition";
+
+  private static final String OSGI_CODEBASE = "file:/";
+
+  private static final String PERM_ADMIN_SERVICE_NAME =
+      "org.osgi.service.condpermadmin.ConditionalPermissionAdmin";
+
+  private ServiceTracker<ConditionalPermissionAdmin, ConditionalPermissionAdmin> permAdminTracker;
+
+  @Override
+  public void start(BundleContext bundleContext) throws Exception {
+    System.setProperty("/", File.separator);
+    permAdminTracker = new ServiceTracker<>(bundleContext, PERM_ADMIN_SERVICE_NAME, null);
+    permAdminTracker.open();
+
+    ConditionalPermissionAdmin conditionalPermissionAdmin = permAdminTracker.getService();
+    String policyDir = SecurityActions.getSystemProperty("ddf.home") + File.separator + "security";
+    if (policyDir.startsWith("=")) {
+      policyDir = policyDir.substring(1);
+    }
+    File policyDirFile = new File(policyDir);
+    List<ParsedPolicy> parsedPolicies = new ArrayList<>();
+    for (File file : Objects.requireNonNull(policyDirFile.listFiles())) {
+      parsedPolicies.add(new Parser(false).parse(new FileReader(new File(file.getAbsolutePath()))));
```

Comment:
❓ What is the point in doing `new File(file.getAbsolutePath())`?
❗️ The FileReader should be properly closed or why not use `Parser.parse(File)`?